sound/soc/sof/ipc4-topology.c

```diff
@@ -1105,11 +1111,57 @@ static int sof_ipc4_init_audio_fmt(struct snd_sof_dev *sdev,
 		sof_ipc4_dbg_audio_format(sdev->dev, &available_fmt->input_pin_fmts[i], 1);
 	}
 
-	if (available_fmt->num_output_formats && i < available_fmt->num_output_formats)
-		base_config->obs = available_fmt->output_pin_fmts[i].buffer_size;
+	/* if module has no output_pin_binding specified in topology, use pin 0 output formats */
+	if (!output_pin_binding) {
+		/*
+		 * if the module only supports 1 output format, copy that buffer size and return
+		 * the index 0. Otherwise, pick the buffer size from the output format matching
+		 * the input format index
+		 */
+		if (available_fmt->num_output_formats == 1) {
+			base_config->obs = available_fmt->output_pin_fmts[0].buffer_size;
+			return 0;
+		}
+
+		if (available_fmt->num_output_formats && i < available_fmt->num_output_formats)
+			base_config->obs = available_fmt->output_pin_fmts[i].buffer_size;
```

Comment:
... here there's no reference to an input format, and the value i comes from a for (i = 0; i < pin_fmts_size; i++) {

I really don't see what the reference to 'input' actually refers to.